### ISSUE TITLE
Tracking issue: context-aware completion

### ISSUE THREAD

**Issue by yannham** [tweag/nickel#1588]
Description WIP.

Related:
 - #1514 
 - #1499 
 - #1095 
 - #877

**Comment by yannham** [comment]
@jneem I wonder how far we are into context-aware completion, as you seem to make very good progress. Is there any task missing here that should be added to the task list? Or should we close this, considered the first phase done?

**Comment by jneem** [comment]
Probably it makes sense to add #1613 here, but I think the most important tasks are already done. For example, it isn't clear that we event want #1095.

**Comment by yannham** [comment]
Right, I put it in the "related" section and not in the task list, but even then, it doesn't really seem to be related in anyway (beside being about "completion"). Let's get it out of the list. For #1613, while it's clearly a nice improvement, I wonder if it's strictly required though to complete this tracking issue - it seems that context-aware completion is already there and working. Do you think #1613 is a nice addition, or something more fundamental that we need to ship or the completer would feel incomplete? If it's the former, I think we can close this tracking issue and organize follow-up work from other tracking issues.

**Comment by yannham** [comment]
Cool, then I think we can probably close :slightly_smiling_face:
